File: adventofcode/Day1.js
var fs = require('fs');
var readline = require('readline');

var readable = fs.createReadStream(__dirname + '/test.in', {
    encoding: 'utf8',
    highWaterMark: 16 * 1024
});
var interface = readline.createInterface({
  input: readable,//process.stdin
  terminal: false
});
interface.on('line', function(line){
    var count = minus1 = 0;
    var found = false;
    for(var i=0; i<line.length; i++) {
        var c = line[i];
        count += (c === '(') ? 1 : -1;
        if(count === -1 && !found) {
            minus1 = i;
            found = true;
        }
    }
    console.log(count, minus1);
});

File: adventofcode/Day1.py
def solve():
    x = interface.r()
    count = minus1 = 0
    found = False
    for i in range(len(x)):
        c = x[i]
        count += 1 if c == '(' else -1;
        if count == -1 and not found:
            found = True
            minus1 = i
    interface.w(count, minus1)

class interface:
    @staticmethod
    def w(s):
        fout.write(s + "\n")
        #print(s)
    @staticmethod
    def r():
        return fin.readline().rstrip()
        #return input()

fin = open('test.in', 'r')
solve()
fin.close()
